### PR TITLE
Fix label info dispatching

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
resolves [ITEP-71658](https://jira.devtools.intel.com/browse/ITEP-71658)

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/open-edge-platform/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/open-edge-platform/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/open-edge-platform/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/model/base.py [hunk 1]
```diff
@@ -410,8 +410,7 @@ def load_state_dict_incrementally(self, ckpt: dict[str, Any], *args, **kwargs) -
             msg = "Checkpoint should have `label_info`."
             raise ValueError(msg, ckpt_label_info)
 
-        if isinstance(ckpt_label_info, dict):
-            ckpt_label_info = LabelInfo(**ckpt_label_info)
```
Comment (sovrasov) [inline src/otx/core/model/base.py:414]: Could you add a snapshot loading test for sseg as well?

Comment (kprokofi) [reply src/otx/core/model/base.py:414]: it is not about loading old snapshot rather than dispatching label info while loading checkpoint for fine-tuning. I made changes in "load_state_dict", but forgot about incremental learning. Anyway, I added a test to check that case.

Comment (kprokofi) [reply src/otx/core/model/base.py:414]: https://github.com/open-edge-platform/training_extensions/pull/4443/commits/55041e903e46c12c3bd772ffa3ffa369cb791832

**Review by sovrasov — APPROVED** [review]
